src/components/Patterns.ts

```diff
@@ -58,6 +58,8 @@ const text = (regexp: RegExp, description: string): TextPattern => ({
 })
 
 export const Patterns = {
+  year: numeric(/[12][0-9]{3}/, 'Input should be a valid year', '####', '_'),
```

Comment:
This regex allows for numbers in the future like 2055 and similar. Could validate instead by using 
`year < (new Date().getFullYear())` or similar

Reply:
react-hook-form's controller supports min/max ranging, so used that.